### PR TITLE
Delete finalizers from all namespaces and serverless secrets

### PR DESCRIPTION
**Pull request by suleymanakbas91**
**Description**

Changes proposed in this pull request:

- Delete finalizers from all namespaces and serverless secrets

**Related issue(s)**

See kyma-project/kyma#11298

### CODE REVIEW

**Review by tgorgol — DISMISSED** [review]

**Review by tgorgol — DISMISSED** [review]

File: parallel-install/pkg/deployment/deletion.go [hunk 1]
```diff
@@ -266,54 +266,54 @@ func (i *Deletion) deleteKymaNamespaces(namespaces []string) error {
...
 				}
+			}
 
-				//HACK: Delete finalizers of leftover Custom Resources
-				crds, err := i.apixClient.CustomResourceDefinitions().List(context.Background(), metav1.ListOptions{})
-				if err != nil {
-					errorCh <- err
-				}
+			//HACK: Delete finalizers of leftover Custom Resources
+			crds, err := i.apixClient.CustomResourceDefinitions().List(context.Background(), metav1.ListOptions{})
```
Comment (strekm) [inline parallel-install/pkg/deployment/deletion.go:286]: i think that only crds installed by kyma should be listed here. we label our CRDs

File: parallel-install/pkg/deployment/deletion.go [hunk 2]
```diff
@@ -266,54 +266,54 @@ func (i *Deletion) deleteKymaNamespaces(namespaces []string) error {
...
-						}
-						if customResourceList != nil {
-							for _, cr := range customResourceList.Items {
-								cr.SetFinalizers(nil)
-								_, err := i.dClient.Resource(customResource).Namespace(ns).Update(context.Background(), &cr, metav1.UpdateOptions{})
-								if err != nil {
-									errorCh <- err
-								}
-								i.cfg.Log.Infof("Deleted finalizer from %s: %s", cr.GetKind(), cr.GetName())
+					customResourceList, err := i.dClient.Resource(customResource).Namespace(ns).List(context.Background(), metav1.ListOptions{})
```
Comment (strekm) [inline parallel-install/pkg/deployment/deletion.go:299]: is there a way to list only these CRs which have finalizers not null?

File: parallel-install/pkg/deployment/deletion.go [hunk 3]
```diff
@@ -266,54 +266,54 @@ func (i *Deletion) deleteKymaNamespaces(namespaces []string) error {
...
+					if err != nil && !apierr.IsNotFound(err) {
+						errorCh <- err
+					}
+					if customResourceList != nil {
+						for _, cr := range customResourceList.Items {
```
Comment (strekm) [inline parallel-install/pkg/deployment/deletion.go:304]: or maybe it's enough to check here if there's finalizer to remove and update only then

**Review by strekm — APPROVED** [review]
/meow

**Pull request merged**